### PR TITLE
Programmer fpga_config, flash and OTP tcl return

### PR DESCRIPTION
**Pull request by guanyung-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

Added tcl status returned to programmer fpga_config, flash and otp command execution. It returns TCL_ERROR whenever the programming failed.

Whenever TCL_ERROR return from fpga_config, flash or otp command, the Programmer GUI will stop the subsequent programming task
image

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [x] Library: Programmer
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/Configuration/CFGCompiler/CFGCompiler.cpp [hunk 1]
```diff
@@ -73,7 +74,7 @@ bool CFGCompiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
               cfgcompiler->m_cmdarg.tclOutput);
           cfgcompiler->m_cmdarg.tclOutput.clear();
         }
-        return result;
+        return result && cfgcompiler->m_cmdarg.tclStatus == TCL_OK;
```
Comment (KochynVolodymyr) [inline src/Configuration/CFGCompiler/CFGCompiler.cpp:77]: This condition doesn't work. Please use this:
```
if ((result != TCL_OK) || (cfgcompiler->m_cmdarg.tclStatus != TCL_OK))
  return TCL_ERROR;
return TCL_OK;
```

File: src/Configuration/CFGCompiler/CFGCompiler.cpp [hunk 2]
```diff
@@ -90,7 +91,7 @@ bool CFGCompiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
...
```
Comment (KochynVolodymyr) [inline src/Configuration/CFGCompiler/CFGCompiler.cpp:94]: This condition doesn't work. Please use this:
```
if ((result != TCL_OK) || (cfgcompiler->m_cmdarg.tclStatus != TCL_OK))
  return TCL_ERROR;
return TCL_OK;
```

File: src/Configuration/CFGCompiler/CFGCompiler.cpp (hunk 1 above)
Comment (guanyung-rs) [reply src/Configuration/CFGCompiler/CFGCompiler.cpp]: I fixed it.

File: src/Configuration/CFGCompiler/CFGCompiler.cpp (hunk 2 above)
Comment (guanyung-rs) [reply src/Configuration/CFGCompiler/CFGCompiler.cpp]: I fixed it using your suggestion

**Comment by guanyung-rs** [comment]
The code coverage drops 0.01%. It doesn't have any unit test now. The unit tests is hard to implement for the CFGCompiler class that inherit Compiler class as it has many concrete object that need to be setup for the unit testing.
@KochynVolodymyr @ravic-rs Do you have any idea to get the COV passed?

**Pull request merged**